lib/miq_automation_engine/models/miq_ae_browser.rb

```diff
@@ -0,0 +1,139 @@
+class MiqAeBrowser
+  attr_accessor :user
+
+  # Automate Datastore Browser starting at MiqAeDomain, down to the MiqAeInstances and MiqAeMethods.
+
+  # Searches directory ala LDAP style
+  #   Starting point of searches using fqname  (dn)
+  #   Supporing scope base, one and sub
+  # Returns array of objects
+  #
+  # Supporting options:
+  #   :serialize
+  #   :state_machines
+  #
+  # :serialize returns the Hash serialized objects including fqname, domain_fqname and klass
+  # i.e.
+  #   {
+  #     "fqname": "/ManageIQ/System/Request",
+  #     "domain_fqname": "/System/Request",       # fqname relative to /domain
+  #     "klass": "MiqAeClass",
+  #     "id": 75,
+  #     "description": "Automation Requests",
+  #     "name": "Request",
+  #     "created_on": "2015-12-09T20:56:44Z",
+  #     "updated_on": "2015-12-09T20:56:44Z",
+  #     "namespace_id": 41,
+  #     "updated_by": "system"
+  #   }
+  #
+  # with :state_machines true, only the objects belonging to a sub-tree that
+  # includes state machine entrypoints are returned.
+
+  SCOPE_BASE = "base".freeze
+  SCOPE_ONE  = "one".freeze
+  SCOPE_SUB  = "sub".freeze
+  SCOPES = [SCOPE_BASE, SCOPE_ONE, SCOPE_SUB].freeze
```

Comment:
np, base is getting just the object where they start the search, one is getting that object plus the direct children, sub, is that object and the expand tree below it.